autosubmit/config/configcommon.py

```diff
@@ -550,14 +550,42 @@ def _normalize_default_section(self, data_fixed: dict) -> None:
 
     @staticmethod
     def _normalize_wrappers_section(data_fixed: dict) -> None:
+        """Normalize the WRAPPERS section to a consistent format so there is no issues during runtime.
+        :param data_fixed: The input data dictionary to normalize.
+        """
         wrappers = data_fixed.get("WRAPPERS", {})
         for wrapper, wrapper_data in wrappers.items():
             if isinstance(wrapper_data, dict):
-                jobs_in_wrapper = wrapper_data.get("JOBS_IN_WRAPPER", "")
-                # if it is a list in string format (due to "%" in the string).
-                if "[" in jobs_in_wrapper:
-                    jobs_in_wrapper = jobs_in_wrapper.strip("[]").replace("'", "").replace(" ", "").replace(",", " ")
-                data_fixed["WRAPPERS"][wrapper]["JOBS_IN_WRAPPER"] = jobs_in_wrapper.upper()
+                jobs_in_wrapper = wrapper_data.get("JOBS_IN_WRAPPER", None)
+                if jobs_in_wrapper:
+                    if not isinstance(jobs_in_wrapper, list) and not isinstance(jobs_in_wrapper, str):
+                        raise AutosubmitCritical(
+                            f"JOBS_IN_WRAPPER in WRAPPERS.{wrapper} must be a list or a string",
+                            7014
+                        )
+                if isinstance(jobs_in_wrapper, str):
+                    # if it is a list in string format (due to "%" in the string).
+                    if "[" in jobs_in_wrapper:
+                        jobs_in_wrapper = jobs_in_wrapper.strip("[]").replace("'", "").replace(" ", "").replace(",", " ")
+                    if "," in jobs_in_wrapper:
+                        jobs_in_wrapper = jobs_in_wrapper.split(",")
+                    elif "&" in jobs_in_wrapper:
+                        jobs_in_wrapper = jobs_in_wrapper.split("&")
+                    else:
+                        jobs_in_wrapper = jobs_in_wrapper.split()
+                sanitazed_jobs_in_wrapper = [job.upper().strip(" ,") for job in jobs_in_wrapper]
+                if jobs_in_wrapper is not None:
+                    for element in sanitazed_jobs_in_wrapper:
+                        if not isinstance(element, str):
```

Comment:
Is that possible? I think because `sanitized_jobs_in_wrapper` is created with `[str.upper().strip() ...]`, that means the type is `list[str]`.

The elements can be only strings? No `None`'s, nor other objects, I think? I think this whole `_normalize_wrappers_section` function can be unit tested, which is faster. And I **think** trying to reach this `if` wouldn't be possible, confirming it's safe to remove...